cluster/manifests/z-kubenurse/service.yaml

```diff
@@ -0,0 +1,19 @@
+---
+# Source: kubenurse/templates/service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    application: kubernetes
+    component: kubenurse
+  name: "kubenurse"
+  namespace: kubenurse
+spec:
+  ports:
+    - name: http
+      port: 8080
+      protocol: TCP
+      targetPort: 8080
+  selector:
+    application: kubernetes
+    component: kubenurse
```

Comment:
What is the purpose of Service (and Ingress)?

Service for a daemonset is generally not desired because it means we need to track ALL pods in the iptable rules of all nodes. So the bigger the cluster, the bigger per node iptable. We have hit memory limits in flannel because of this in the past, so if we don't strictly need the service, it's better to not have it.

Reply:
kubenurse uses the service for one of its tests [(reach to self through service)](https://github.com/postfinance/kubenurse#me-service)

> Checks if the kubenurse is reachable at the /alwayshappy endpoint through the Kubernetes service. The address is provided by the environment variable KUBENURSE_SERVICE_URL that could look like http://kubenurse.mynamespace.default.svc:8080. This also verifies a working kube-proxy setup.

Reply:
similarly, the ingress is used to verify it works upstream (DNS resolution and routing)

Reply:
> if we don't strictly need the service (& ingress), it's better to not have it
I think the bare minimum requirement is that we check pod-to-pod communication, although these checks are useful, maybe we can omit them for now?

Reply:
> I think the bare minimum requirement is that we check pod-to-pod communication, although these checks are useful, maybe we can omit them for now?

I agree. Better to start small especially when we know of the problems of the service for daemonset setup.

Reply:
in progress
[postfinance/kubenurse/pull#54](https://github.com/postfinance/kubenurse/pull/54)